src/Configuration/Annotation/Embedded.php

```diff
@@ -4,16 +4,23 @@
 
 namespace Hateoas\Configuration\Annotation;
 
+use JMS\Serializer\Annotation\AnnotationUtilsTrait;
+use Symfony\Contracts\Service\Attribute\Required;
+
 /**
  * @Annotation
  * @Target("ANNOTATION")
  */
+#[\Attribute(0)]
 class Embedded
 {
+    use AnnotationUtilsTrait;
+
     /**
      * @Required
      * @var mixed
      */
+    #[Required]
```

Comment:
The `@Required` annotation does not reference a symfony class but [`Doctrine\Common\Annotations\Annotation\Required`](https://github.com/doctrine/annotations/blob/2.0.x/lib/Doctrine/Common/Annotations/Annotation/Required.php).

I suggest to remove `#[Required]` everywhere in this PR.

Reply:
Yes, I was wrong to use the Symfony `Required` class which is about the service auto-wiring.  So, it has nothing to do with the Doctrine annotation purpose. 

I will remove `#[Required]`